src/Makefile

```diff
@@ -1473,7 +1473,8 @@ ALL_LIBS = \
 	   $(RUBY_LIBS) \
 	   $(PROFILE_LIBS) \
 	   $(SANITIZER_LIBS) \
-	   $(LEAK_LIBS)
+	   $(LEAK_LIBS) \
+	   $(HUNSPELL_LIBS)
```

Comment:
You're missing `$(HUNSPELL_CFLAGS)` earlier in the Makefile.  https://github.com/jamessan/vim/commit/09f1bf8d4c721f1b4fe50c22a345d4cf5129e7a3 should cover what needs to be done.

Reply:
Good point. I have not need it, but it is probably The Right Thing™ to do. Added in 387a0a17d (and built as https://copr.fedorainfracloud.org/coprs/mcepl/vim8/build/691979/).

> If Bram is receptive to this work, I can actually dedicate some time to it.

That would be awesome, I gladly admit I have no idea about what I am doing with C programming, so any help is more than welcome.